packages/react-router/lib/deprecations.ts

```diff
@@ -4,7 +4,7 @@ import type { FutureConfig as RenderFutureConfig } from "./components";
 const alreadyWarned: { [key: string]: boolean } = {};
 
 export function warnOnce(key: string, message: string): void {
-  if (!alreadyWarned[message]) {
+  if (process.env.NODE_ENV !== 'production' && !alreadyWarned[message]) {
```

Comment:
We have a build time `__DEV__` constant we can swap out for dev/prod builds to allow the library to remain runtime agnostic

```suggestion
  if (__DEV__ && !alreadyWarned[message]) {
```